libraries/GCS_MAVLink/GCS.h

```diff
@@ -267,6 +268,16 @@ class GCS_MAVLINK
     // accessor for uart
     AP_HAL::UARTDriver *get_uart() { return _port; }
 
+#if AP_SCHEDULER_ENABLED
+    // cap the MAVLink message rate. It can't be greater than 0.8 * SCHED_LOOP_RATE
+    uint16_t cap_message_interval(uint16_t interval_ms) {
+        if (interval_ms != 0 && interval_ms*800 < AP::scheduler().get_loop_period_us()) {
```

Comment:
```suggestion
        if (interval_ms*800 < AP::scheduler().get_loop_period_us()) {
```

kind of a given.  Unless your loop period is zero, in which case you have bigger problems.  And you'll still return the right result

Reply:
removed the zero check